web/src/message_edit.js

```diff
@@ -792,9 +792,15 @@ export function end_inline_topic_edit($row) {
 
 function remove_uploads_from_row(row_id) {
     const uploads_for_row = upload.upload_objects_by_message_edit_row.get(row_id);
+    if (!uploads_for_row) {
+        return;
+    }
     // We need to cancel all uploads, reset their progress,
     // and clear the files upon ending the edit.
-    uploads_for_row?.cancelAll();
+    upload.deactivate_upload(uploads_for_row, {
+        mode: "edit",
+        row: row_id,
+    });
     // Since we removed all the uploads from the row, we should
     // now remove the corresponding upload object from the store.
     upload.upload_objects_by_message_edit_row.delete(row_id);
```

Comment:
This looks good, but I think you should add a second commit changing the interface between modules such that `upload_objects_by_message_edit_row` isn't exported; it should be the case that one can just call `upload.deactivate_upload` passing `edit` and the `row_id`, and that function takes care of looking up any upload objects and cleaning them.

We'd need to change `start_edit_with_content` slightly too, but that also seems like an improvement to the abstractions: The upload module can be solely responsible for maintaining its upload objects and `upload_objects_by_message_edit_row`.

Reply:
Handled the export of objects `upload_objects_by_message_edit_row` and `compose_upload_object` via commit cfe1f5dfda2e2c09267dd1ed44b055146adf34d6. After this commit, we don't export any other objects in `upload.js`.